lib/ui/lerp.dart

```diff
@@ -6,15 +6,23 @@
 
 part of dart.ui;
 
-/// Linearly interpolate between two numbers.
-// TODO(cbracken): Consider making a and b non-nullable.
-//                 https://github.com/flutter/flutter/issues/64617
+/// Linearly interpolate between two numbers, `a` and `b`, by an extrapolation
+/// factor `t`.
+///
+/// When both `a` and `b` are null, infinite, or NaN, `a` is returned.
+/// Otherwise, if `a`, `b`, and `t` are required to be finite or null, and the
+/// result of `a + (b - a) * t` is returned, where nulls are defaulted to 0.0.
 double? lerpDouble(num? a, num? b, double t) {
-  if (a == null && b == null)
-    return null;
-  a ??= 0.0;
-  b ??= 0.0;
-  return a + (b - a) * t as double;
+  if (a == b)
+    return a as double?;
+  final double from = a as double? ?? 0.0;
+  final double to = b as double? ?? 0.0;
+  if (from.isNaN && to.isNaN)
+    return double.nan;
+  assert(from.isFinite, 'a must be finite');
+  assert(to.isFinite, 'b must be finite');
+  assert(t.isFinite, 't must be finite');
```

Comment:
these are confusing messages given that you do support a and b being NaN etc. :-)
Maybe "Cannot interpolate between finite and non-finite values" or some such?

Reply:
Done.